test/test.js

```diff
@@ -52,4 +75,26 @@ describe('onclickoutside hoc', function() {
       assert(e, "component was not wrapped");
     }
   });
+
+  it('should not call handleClickOutside if this.props.disableOnClickOutside() is called, until this.props.enableOnClickOutside() is called.', function() {
+    var element = React.createElement(WrappedComponent);
+    var component = TestUtils.renderIntoDocument(element);
+    document.dispatchEvent(new Event('mousedown'));
+    var instance = component.getInstance();
+    assert(instance.state.timesHandlerCalled === 1, "handleClickOutside called");
+
+    try {
+      externalToggleEnableClickOutside(false);
```

Comment:
I implemented this change. Shaves off the need for componentDidMount. I made the test seem cleaner.